src/main/java/net/neoforged/neoforge/event/entity/living/ArmorHurtEvent.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.living;
+
+import java.util.EnumMap;
+import java.util.Map;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.ICancellableEvent;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Fired when a {@link LivingEntity}'s armor is dealt damage in {@link LivingEntity#doHurtEquipment(DamageSource, float, EquipmentSlot...) doHurtEquipment}.
+ * <p>
+ * This event is {@link ICancellableEvent cancelable}. Cancelling this event will prevent all damage to armor.
+ * <p>
+ * This event does not have a result.
+ * <p>
+ * This event is fired on the {@link NeoForge#EVENT_BUS}
```

Comment:
Similar comment as `EntityInvulnerablityCheckEvent` - though since this event is `ICancellableEvent`, the documentation on what cancelling the event does should be placed in an override of `ICancellableEvent#setCanceled`.